### ISSUE TITLE
Feature Request: Joliet and Rock Ridge extensions supporting

### ISSUE THREAD

**Issue by vdmvoro** [diskfs/go-diskfs#73]
Good day
Hope you are doing well.

For now: Joliet and Rock Ridge extensions are not supported (I mean reading or writing).
As I see you have a plans:

- Joliet extensions to iso9660
- Rock Ridge sparse file support - supports the flag, but not yet reading or writing

Do you plan to write additional methods for this in near future? Maybe you know some ETA?

**Comment by deitch** [comment]
At some point, yes. squashfs writing and ext4 writing have been bogged down for a while. I would like to see Joliet and RR in as well. 

Would be more than happy to take contributions/Pull Requests. If not, it is just a question of finding time.

**Comment by davidnewhall** [comment]
TL;DR: I found 1 [old library](https://github.com/dgodd/iso9660/tree/master) that has Rock Ridge support, and 1 that [supports UDF](https://github.com/mogaika/udf). I found none that support Joliet.

Recently I did a lot of googling and GitHub searching to find libraries that allow me to extract ISO images. The real problem is most of my users download random stuff and the image types are all over the place. I quickly found that there aren't many libraries out there, and none of them support _everything_ yet. Maybe one day we'll get there. I just wanted to share some of my research in case anyone finds time to write the implementation(s). I'd put more time into it, but I'm far from an expert on file systems, and a lot of this is very new to me. I work best from pointers and examples, so if there's anything to add, please throw it in here.

The info I found is spread through [a couple comments on this issue](https://github.com/Unpackerr/unpackerr/issues/264#issuecomment-1527016993).

**Comment by deitch** [comment]
I could point you at all of the structures for writing disk images and filesystems here, but RR and Joliet are extensions to an existing filesystem format, not a new one. The references all are linked in [this brief doc](https://github.com/diskfs/go-diskfs/blob/master/filesystem/iso9660/doc.go)

**Comment by luthermonson** [comment]
@deitch this can be closed
